source/beamline/IDEAS/IDEAS13ElementGeDetector.cpp

```diff
@@ -17,6 +18,27 @@ IDEAS13ElementGeDetector::IDEAS13ElementGeDetector(const QString &name, const QS
 	acquireTimeControl_ = new AMSinglePVControl("Integration Time", "dxp1608-B21-13:PresetReal", this, 0.001);
 	elapsedTimeControl_ = new AMReadOnlyPVControl("Elapsed Time", "dxp1608-B21-13:ElapsedReal", this);
 
+	// Detector functionality
+	ge13ElementRealTimeControl_ = new AMReadOnlyPVControl("13-el Ge Real Time", "dxp1608-B21-13:ElapsedReal", this);
+	ge13ElementRealTime_ = new AMBasicControlDetectorEmulator("13E_dwellTime", "13-element Ge dwell time", ge13ElementRealTimeControl_, 0, 0, 0, AMDetectorDefinitions::ImmediateRead, this);
+
+    // Energy Peaking Time
+    peakingTimeControl_ = new AMSinglePVControl("Peaking Time", "dxp1608-B21-13:EnergyPkTime", this, 0.01);
+
+    // preamp Gain controls for elements 1-12
+    preampGainControl1_ = new AMPVControl("13-el Ge Preamp Gain 1","dxp1608-B12-13:dxp1:PreampGain_RBV","dxp1608-B12-13:dxp1:PreampGain", QString(), this, AMCONTROL_TOLERANCE_DONT_CARE);
```

Comment:
This could be replaced by a list (i.e.: `preampGainControls_`) and then you culd have made a simple loop such as:

```
for (int i = 0; i < 12; i++){
     preampGainControls_ << new AMPVControl(QString("13-el Ge Preamp Gain %1").arg(i+1), QString(dxp1608-B12-13:dxp%1:PreampGain_RBV").arg(i+1), "dxp1608-B12-13:dxp%1:PreampGain", QString(), this);
}
```